#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_r1352_firefox_workaround.dpatch by Lennart Poettering <mzcnihpbageby@0pointer.de>
##
## DP: work around bug in firefox which apparently misuses access() as NULL
## DP: pointer test. Original patch by "alon".

@DPATCH@
diff -urNad pulseaudio~/src/utils/padsp.c pulseaudio/src/utils/padsp.c
--- pulseaudio~/src/utils/padsp.c	2006-08-18 23:29:26.000000000 +0200
+++ pulseaudio/src/utils/padsp.c	2006-10-19 00:13:47.082198231 +0200
@@ -2121,6 +2121,13 @@
 }
 
 int access(const char *pathname, int mode) {
+
+    if (!pathname) {
+        /* Firefox needs this. See #27 */
+        errno = EFAULT;
+        return -1;
+    }
+    
     debug(DEBUG_LEVEL_VERBOSE, __FILE__": access(%s)\n", pathname);
 
     if (strcmp(pathname, "/dev/dsp") != 0 &&
